lib/promscrape/scrapework.go

```diff
@@ -996,7 +1021,7 @@ func appendRow(dst []parser.Row, metric string, value float64, timestamp int64)
 	})
 }
 
-func (wc *writeRequestCtx) addRows(cfg *ScrapeWork, rows []parser.Row, timestamp int64, needRelabel bool) {
+func (wc *writeRequestCtx) tryAddRows(cfg *ScrapeWork, rows []parser.Row, timestamp int64, needRelabel bool) error {
```

Comment:
I think, it's better to keep `addRows` naming and just add error. 

 `try` prefix is better to use with some boolean return result.

Reply:
renamed